core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -49,16 +49,16 @@ class KryoSerializer(conf: SparkConf)
   with Logging
   with Serializable {
 
-  private val bufferSizeMb = conf.getDouble("spark.kryoserializer.buffer.mb", 0.064)
+  private val bufferSizeMb = conf.getSizeAsKb("spark.kryoserializer.buffer", "64k").toDouble/1000.0d
```

Comment:
I think this is incorrect. If the user specifies `0.064` as before this will be interpreted as `0.064kb` even though they mean `0.064mb`